forte/pipeline.py

```diff
@@ -466,6 +470,28 @@ def save(self, path: str):
         with open(path, "w", encoding="utf-8") as f:
             yaml.safe_dump(self._dump_to_config(), f)
 
+    def export(self, name: str = "pipeline") -> Optional[str]:
+        r"""Export pipeline to FORTE_EXPORT_PATH
+        FORTE_EXPORT_PATH is an environment variable
+        The pipeline is saved by :meth:`~forte.pipeline.Pipeline.save`
+
+        Args:
+            name: Export name of the pipeline. Default is `pipeline`.
+
+        Returns:
+            Optional[str]: Export path of pipeline config YAML.
+        """
+        export_path: Optional[str] = os.environ.get("FORTE_EXPORT_PATH")
+        if export_path:
+            if not os.path.exists(export_path):
+                os.makedirs(export_path, exist_ok=True)
+            export_path = os.path.join(
+                export_path, f"{name}-{self._export_count}.yml"
```

Comment:
Better to just require the `name` argument (not default it to `pipeline`) and raise a `ValueError` if there is a conflict. adding a suffix will be unintuitive to users who write the lines
```
pipeline.export("my-pipeline")
pipeline.export("my-new-pipeline")
```
but pipelines with the names `my-pipeline-1` and `my-new-pipeline-2` show up on the other end

Reply:
D'oh just saw @hunterhector's other comment. Ignore this one.

Reply:
But I think your comment also makes sense.
How about this:
- If user didn't specify an export name, then the default one will be `pipeline-<self._export_count>.yml`
- The `self._export_count` only increments if the pipeline is exported with the default name
- If user provides an export name, then we'll use it. And raise `ValueError` if conflicts.